vnext/Microsoft.ReactNative/CoreApp.cpp

```diff
@@ -0,0 +1,306 @@
+      if (auto nm_value = nativeModules.GetAt(index); nm_value.ValueType() == JsonValueType::Object) {
+        auto nm = nm_value.GetObject();
+
+        const wchar_t *container{nullptr};
+        SetFromJson(nm, L"moduleContainer", container);
+
+        const wchar_t *factoryName{nullptr};
+        SetFromJson(nm, L"factory", factoryName);
+
+        HMODULE mod{};
+        if (container != nullptr) {
+          mod = reinterpret_cast<HMODULE>(WINRT_IMPL_LoadLibraryW(container));
+        } else {
+          mod = GetModuleHandle(nullptr);
+        }
+
+        if (mod) {
+          if (auto factory = reinterpret_cast<void *(__cdecl *)()>(
+                  WINRT_IMPL_GetProcAddress(mod, winrt::to_string(factoryName).c_str()))) {
+            auto packageProviderAbi = factory();
+            app->packageProvidersAbi[countAdded++] = packageProviderAbi;
+          } else {
+            assert(false && "Could not find native module factory function");
+          }
+        } else {
+          assert(false && "Could not find native module library");
+        }
+      }
+    }
+    app->packageProvidersAbiCount = countAdded;
+  }
+}
+
+static void SetProperties(const JsonObject &json, RNCoreApp *app) {
+  react::IReactPropertyBag bag{nullptr};
+  if (auto v = json.TryLookup(L"properties"); v && v.ValueType() == JsonValueType::Object) {
+    auto props = v.GetObject();
+    bag = react::ReactPropertyBagHelper::CreatePropertyBag();
+    for (const auto &prop : props) {
+      auto name = prop.Key();
+      auto lastDot = std::find(name.rbegin(), name.rend(), L'.');
+      std::wstring nsName;
+      std::wstring localName;
+      if (lastDot != name.rend()) {
+        nsName = std::wstring{name.begin(), static_cast<uint64_t>(name.rend() - lastDot - 1)};
+        localName = std::wstring{lastDot.base()};
+      } else {
+        localName = name;
+      }
+      winrt::hstring nsNameH(nsName);
+      winrt::hstring localNameH(localName);
+      auto propName = react::ReactPropertyName{nsNameH, localNameH};
+      auto value = prop.Value();
+      switch (value.ValueType()) {
+        case JsonValueType::String:
+          bag.Set(propName.Handle(), winrt::box_value(value.GetString()));
+          break;
+        case JsonValueType::Boolean:
+          bag.Set(propName.Handle(), winrt::box_value(value.GetBoolean()));
+          break;
+        case JsonValueType::Number:
+          bag.Set(propName.Handle(), winrt::box_value(value.GetNumber()));
+          break;
+        case JsonValueType::Null:
+          bag.Set(propName.Handle(), nullptr);
+          break;
+        default:
+          assert(false && "not yet implemented");
+          break;
+      }
+    }
+  }
+
+  if (bag) {
+    winrt::copy_to_abi(bag, app->propertiesAbi);
+  }
+}
+
+extern "C" NORETURN void __cdecl RNStartCoreAppFromConfigJson(wchar_t const *configJson, coreAppCallback launched) {
+  auto app = new RNCoreApp();
```

Comment:
>app [](http://example.com/codeflow?start=7&length=3)

Allocating and never freeing this variable seem "unusual". Below we use shared_ptr in the same context.
We can also use unique_ptr, or better yet allocate it on the call stack. The end result - no 'naked' new.